frontend/model/contracts/voting/proposals.js

```diff
@@ -122,10 +123,41 @@ const proposals = {
         [RULE_DISAGREEMENT]: { threshold: 2 }
       }
     },
-    [VOTE_FOR]: function (state, { proposalHash, passPayload }) {
-      console.error('unimplemented!')
-      // TODO: unsubscribe from their mailbox and identity contract
-      //       call commit('removeContract'), etc.
+    [VOTE_FOR]: async function (state, { proposalHash, passPayload }) {
+      const proposal = state.proposals[proposalHash]
+      const { member, memberID, groupID } = proposal.data.proposalData
+      proposal.status = STATUS_PASSED
+      proposal.payload = passPayload
+
+      const data = {
+        member,
+        memberID,
+        groupID,
+        proposalHash,
+        proposalPayload: passPayload
+      }
+      const message = { data, meta: proposal.meta }
+      sbp('gi.contracts/group/removeMember/process', message, state)
+
+      removeMemberSideEffect(data)
+      // // NOTE_1: I tried to call selector sideEffect manually but it didnt work
+      // // TypeError: selectors[selector] is not a function - sbp.js
```

Comment:
Please read this code comment.

Reply:
yeah it's a bit confusing how it was defined in `Contract.js`, but if you have a look there you'll see it needs to be `'gi.contracts/group/removeMember/process/sideEffect'`. I'm considering rewriting as you've done here, as I think that might be more intuitive, feel free to open an issue for that if you'd like.

Reply:
Oh, okay, it makes sense now, because `sideEffect` is part of process. I'm okay with keeping it as it is for now.